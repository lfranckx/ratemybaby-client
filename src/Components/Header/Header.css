nav {
    padding: 10px 40px;
    display: flex;
    justify-content: space-around;
    justify-content: space-between;
    align-items: center;
}

nav a {
    color: #444;
    text-decoration: none;
}

h1 {
    margin: 0;
    font-size: 3.5rem;
    font-weight: 500;
    font-family: 'ProximaNova';
    color: white;
}

h2 {
    color: #444;
    font-size: 2rem;
    opacity: 0.9;
    margin: 0;
}

#title-logo {
    display: inherit;
}

#logo {
    position: relative;
    top: 8px;
    height: 45px;
    width: 40px;
    margin-right: 5px;
}

#todler {
    color: rgb(249, 103, 103);
}

.Header__not-logged-in a:last-child {
    background: white;
    color: rgb(249, 103, 103);
    border: solid 2px rgb(249, 103, 103);
    border-radius: 40px;
    padding: 8px 20px;
    border-radius: 5px;
    margin-left: 10px;
    /* font-size: large; */
    font-weight: bold;
}

.Header__not-logged-in a:last-child:hover {
    background: rgb(249, 103, 103);
    color: white;
}

.Header__logged-in a {
    background: white;
    color: rgb(228, 95, 95);
    border: solid 2px rgb(249, 103, 103);
    border-radius: 40px;
    padding: 8px 20px;
    border-radius: 5px;
    margin-left: 10px;
    font-weight: bold;
}  

.Header__logged-in a:hover {
    background: rgb(249, 103, 103);
    color: white;
}