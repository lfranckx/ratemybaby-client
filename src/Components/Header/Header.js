import React, { Component } from 'react'
import './Header.css'
import { Link } from 'react-router-dom'
import TokenService from '../../Services/token-service'
import IdleService from '../../Services/idle-service'
import Hamburger from '../Hamburger/Hamburger'

export default class Nav extends Component {
    handleLogout = () => {
        TokenService.clearAuthToken()
        /* when logging out, clear the callbacks to the refresh api and idle auto logout */
        TokenService.clearCallbackBeforeExpiry()
        IdleService.unRegisterIdleResets()
    }
    
    renderLogoutLink() {
        return (
            <>
                <nav 
                    aria-label='primary'
                    className="header">
                    <Link id="title-logo" to='/rate'>
                        <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/logo-red.png" 
                            alt="logo" 
                            id="logo" />  
                        <h1>toddler</h1>
                    </Link>
                    <div className='Header__logged-in'>
                        <Link 
                            to='/createprofile'
                        >
                            Create Profile
                        </Link>
                        <Link
                            onClick={this.handleLogout}
                            to='/'>
                            LOG OUT
                        </Link>
                    </div>  
                </nav> 
                <Hamburger />   
            </>    
        )
    }

    renderLoginLink() {
        return (
            <nav className="header">
                <Link id="title-logo" to='/'>
                    <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/logo-red.png" 
                        alt="logo" 
                        id="logo" />  
                    <h1>toddler</h1>
                </Link>
                <div className='Header__not-logged-in'>
                    <Link 
                        id='loginlink'
                        to='/login'>
                        LOGIN
                    </Link>
                </div>
            </nav>
        )
    }

    render() {
        return ( 
                <header>
                    {TokenService.hasAuthToken()
                    ? this.renderLogoutLink()
                    : this.renderLoginLink()}
                </header>
        )
    }
}