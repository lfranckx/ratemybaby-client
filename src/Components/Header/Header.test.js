import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe(`Header component`, () => {

    it(`renders header without crashing`, () => {
        const wrapper = shallow(<Header />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it(`links to log in page`, () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('#loginlink').props().to).toBe('/login');
    });
});