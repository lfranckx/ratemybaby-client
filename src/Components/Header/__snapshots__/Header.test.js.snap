// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders header without crashing 1`] = `
<header>
  <nav
    className="header"
  >
    <Link
      id="title-logo"
      to="/"
    >
      <img
        alt="logo"
        id="logo"
        src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/logo-red.png"
      />
      <h1>
        toddler
      </h1>
    </Link>
    <div
      className="Header__not-logged-in"
    >
      <Link
        id="loginlink"
        to="/login"
      >
        LOGIN
      </Link>
    </div>
  </nav>
</header>
`;
